packages/components/src/Dialog/Dialog.tsx

```diff
@@ -60,79 +36,56 @@ const CloseButton = ({ className }: CloseButtonProps) => {
   );
 };
 
-/**
- * Search for a direct child that can act as title to improve accessibility.
- */
-const addTitleProps = (
-  children: ReactNode,
-  titleProps: HTMLAttributes<HTMLElement>
-) => {
-  const childs = Children.toArray(children);
-
-  const titleIndex = childs.findIndex(
-    child =>
-      isValidElement(child) &&
-      (child.type === Header || child.type === Headline)
-  );
-
-  // No child found that can act as title
-  if (titleIndex < 0) {
-    console.warn(
-      'No child in <Dialog> found that can act as title for accessibility. Please add a <Header> or <Headline> as direct child.'
-    );
-    return children;
-  }
-
-  // If we found a child, add the titleProps to it
-  const titleChild = cloneElement(
-    childs[titleIndex] as ReactElement<any>,
-    titleProps
-  );
-  childs.splice(titleIndex, 1, titleChild);
+// Dialog Headline
+// ---------------
+interface DialogHeadlineProps extends Omit<HeadlineProps, 'slot'> {}
 
-  return childs;
-};
+const DialogHeadline = ({ children }: DialogHeadlineProps) => (
+  <Headline slot="title">{children}</Headline>
+);
 
 // Props
 // ---------------
-export interface DialogChildProps {
-  close: DialogContextProps['close'];
-  titleProps: HTMLAttributes<HTMLElement>;
-}
-
-export interface DialogProps extends AriaDialogProps {
-  children?: ReactNode | ((props: DialogChildProps) => ReactNode);
+export interface DialogProps extends RAC.DialogProps {
```

Comment:
can you remove the `className` and `style` from here? We called it removedProps @OsamaAbdellateef